### PR TITLE
[WEJBHTTP-140] First round of refactorings in TXN part - focus on client side of the protocol

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.redhat.com/browse/WEJBHTTP-140

### CODE REVIEW

**Review by fl4via — APPROVED** [review]
Please have a look at my comment on finals.

File: transaction/src/main/java/org/wildfly/httpclient/transaction/HttpRemoteTransactionHandle.java [hunk 1]
```diff
@@ -157,14 +157,15 @@ public void rollback() throws SecurityException, SystemException {
 
             final CompletableFuture<Void> result = new CompletableFuture<>();
             statusRef.set(Status.STATUS_COMMITTING);
-            ClientRequest cr = new ClientRequest()
-                    .setMethod(Methods.POST)
-                    .setPath(targetContext.getUri().getPath() + TXN_CONTEXT + VERSION_PATH + targetContext.getProtocolVersion()
+
+            RequestBuilder builder = new RequestBuilder().setRequestType(RequestType.UT_ROLLBACK).setVersion(targetContext.getProtocolVersion());
```
[the remark this answers is not in the document]
Comment (ropalka) [reply transaction/src/main/java/org/wildfly/httpclient/transaction/HttpRemoteTransactionHandle.java]: Fixed.

File: transaction/src/main/java/org/wildfly/httpclient/transaction/HttpRemoteTransactionPeer.java [hunk 2]
```diff
@@ -87,13 +86,13 @@ public SubordinateTransactionControl lookupXid(Xid xid) throws XAException {
...
         final CompletableFuture<Xid[]> xidList = new CompletableFuture<>();
 
-        ClientRequest cr = new ClientRequest()
-                .setPath(targetContext.getUri().getPath() + TXN_CONTEXT + VERSION_PATH + targetContext.getProtocolVersion() +
-                        XA_RECOVER_PATH + "/" + parentName)
-                .setMethod(Methods.GET);
-        cr.getRequestHeaders().put(Headers.ACCEPT, XID_LIST + "," + NEW_TRANSACTION);
-        cr.getRequestHeaders().put(RECOVERY_PARENT_NAME, parentName);
-        cr.getRequestHeaders().put(RECOVERY_FLAGS, Integer.toString(flag));
+        RequestBuilder builder = new RequestBuilder().setRequestType(RequestType.XA_RECOVER).setVersion(targetContext.getProtocolVersion());
```
[the remark this answers is not in the document]
Comment (ropalka) [reply transaction/src/main/java/org/wildfly/httpclient/transaction/HttpRemoteTransactionPeer.java]: Fixed.

File: transaction/src/main/java/org/wildfly/httpclient/transaction/HttpRemoteTransactionHandle.java [hunk 3]
```diff
@@ -157,14 +157,15 @@ public void rollback() throws SecurityException, SystemException {
...
+            request.setPath(targetContext.getUri().getPath() + TXN_CONTEXT + VERSION_PATH + targetContext.getProtocolVersion()
                             + UT_ROLLBACK_PATH);
-            cr.getRequestHeaders().put(Headers.ACCEPT, EXCEPTION.toString());
-            cr.getRequestHeaders().put(Headers.CONTENT_TYPE, XID.toString());
-            targetContext.sendRequest(cr, sslContext, authenticationConfiguration, output -> {
-                Marshaller marshaller = targetContext.getHttpMarshallerFactory(cr).createMarshaller();
+            request.getRequestHeaders().put(Headers.ACCEPT, EXCEPTION.toString());
+            request.getRequestHeaders().put(Headers.CONTENT_TYPE, XID.toString());
+            targetContext.sendRequest(request, sslContext, authenticationConfiguration, output -> {
+                Marshaller marshaller = targetContext.getHttpMarshallerFactory(request).createMarshaller();
```
[the remark this answers is not in the document]
Comment (ropalka) [reply transaction/src/main/java/org/wildfly/httpclient/transaction/HttpRemoteTransactionHandle.java:154]: Fixed.

File: transaction/src/main/java/org/wildfly/httpclient/transaction/HttpRemoteTransactionPeer.java [hunk 4]
```diff
@@ -148,12 +139,10 @@ public Xid[] recover(int flag, String parentName) throws XAException {
     public SimpleTransactionControl begin(int timeout) throws SystemException {
         final CompletableFuture<Xid> beginXid = new CompletableFuture<>();
 
-        RequestBuilder builder = new RequestBuilder().setRequestType(RequestType.UT_BEGIN).setVersion(targetContext.getProtocolVersion());
+        RequestBuilder builder = new RequestBuilder().setRequestType(RequestType.UT_BEGIN).setVersion(targetContext.getProtocolVersion()).setTimeout(timeout);
```
[the remark this answers is not in the document]
Comment (ropalka) [reply transaction/src/main/java/org/wildfly/httpclient/transaction/HttpRemoteTransactionPeer.java]: Fixed.

**Pull request merged**